src/uct/ib/base/ib_iface.c

```diff
@@ -502,17 +502,10 @@ void uct_ib_iface_fill_attr(uct_ib_iface_t *iface, uct_ib_qp_attr_t *attr)
 
 #if HAVE_DECL_IBV_EXP_CREATE_QP
     attr->ibv.comp_mask           = IBV_EXP_QP_INIT_ATTR_PD;
-    attr->ibv.pd                  = uct_ib_iface_qp_pd(iface);
+    attr->ibv.pd                  = iface->ops->get_qp_pd(iface);
```

Comment:
if PD flag not set in comp_mask, then set default PD
and set upstream td/exp resd in mlx5_fill_attr